lib/fetch-popular-collection.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+module.exports = function (bus, client, transform) {
+	const fetchPopularCollection = args => {
+		const spec = args.spec;
+		const channel = args.channel;
+		const secrets = channel.secrets || {};
+		let collection;
+
+		const creds = Object.create(null);
+		if (secrets.backlotApiKey && secrets.backlotSecretKey) {
+			creds.apiKey = secrets.backlotApiKey;
+			creds.secretKey = secrets.backlotSecretKey;
+		}
+
+		// First, get the discovery object from Ooyala.
+		return client.getPopularRelated(creds)
+			.then(popular => {
+				if (popular) {
+					// If the popular results exist, cast it to an Oddworks collection.
+
+					// There should only be one popular collection
+					spec.id = 'spec-ooyala-discovery-popular';
+					collection = transform(spec, {name: 'Popular Videos'});
```

Comment:
Thoughts on making `'Popular Videos'` configurable from the spec so this isn't hardcoded?